governance/api.go

```diff
@@ -146,22 +162,166 @@ func (api *GovernanceKlayAPI) GetRewards(num *rpc.BlockNumber) (*reward.RewardSp
 	return reward.GetBlockReward(header, rules, rewardParamSet)
 }
 
-func (api *GovernanceKlayAPI) ChainConfig() *params.ChainConfig {
-	num := rpc.LatestBlockNumber
-	return getChainConfig(api.governance, &num)
-}
+type accumulatedRewards struct {
+	Period         uint     `json:"period"`
+	FirstBlockTime string   `json:"firstBlockTime"`
+	LastBlockTime  string   `json:"lastBlockTime"`
+	FirstBlock     *big.Int `json:"firstBlock"`
+	LastBlock      *big.Int `json:"lastBlock"`
+
+	// TotalMinted + TotalTxFee - TotalBurntTxFee = TotalProposerRewards + TotalStakingRewards + TotalKFFRewards + TotalKCFRewards
+	TotalMinted          *big.Int                    `json:"totalMinted"`
+	TotalTxFee           *big.Int                    `json:"totalTxFee"`
+	TotalBurntTxFee      *big.Int                    `json:"totalBurntTxFee"`
+	TotalProposerRewards *big.Int                    `json:"totalProposerRewards"`
+	TotalStakingRewards  *big.Int                    `json:"totalStakingRewards"`
+	TotalKFFRewards      *big.Int                    `json:"totalKFFRewards"`
+	TotalKCFRewards      *big.Int                    `json:"totalKCFRewards"`
+	Rewards              map[common.Address]*big.Int `json:"rewards"`
+}
+
+// GetRewardsAccumulated returns accumulated rewards data in the range of [start, end).
+// The given block range is divided into many accumulation periods with a given period parameter.
+func (api *GovernanceAPI) GetRewardsAccumulated(start rpc.BlockNumber, end rpc.BlockNumber, period uint64) ([]accumulatedRewards, error) {
```

Comment:
"interval" might be a better term.
`numPeriods = (end - start) / interval`

Besides, is it necessary for this API to provide a multi-period mechanism? Caller can call this API multiple times.

Reply:
removed duration